db/models/user.js

```diff
@@ -44,11 +40,9 @@ function setEmailAndPassword(user) {
   user.email = user.email && user.email.toLowerCase()
   if (!user.password) return Promise.resolve(user)
 
-  return new Promise((resolve, reject) =>
-    bcrypt.hash(user.get('password'), 10, (err, hash) => {
-      if (err) return reject(err)
+  return bcrypt.hash(user.get('password'), 10)
+    .then(hash => {
       user.set('password_digest', hash)
-      resolve(user)
+      return user
```

Comment:
if `user.set` returns a promise for a user — which it may or may not, I don't know offhand — this can become `.then(hash => user.set('password_digest'))`.